[JSC] Use libpas for Structure Heap

Constellation opened the pull request:
#### 857df8d3aa2ecf223b7ac9c5c29d893c369cd4a9
<pre>
[JSC] Use libpas for Structure Heap
<a href="https://bugs.webkit.org/show_bug.cgi?id=290685">https://bugs.webkit.org/show_bug.cgi?id=290685</a>
<a href="https://rdar.apple.com/148155464">rdar://148155464</a>

Reviewed by Yijia Huang.

This patch makes StructureAlignedMemoryAllocator much simpler.

1. We remove IsoAlignedMemoryAllocator parts and just let StructureMemoryManager
   allocate and free memory.
2. In libpas-enabled environment, libpas governs Structure heap, and we
   just use libpas functions to allocate and free memory.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/heap/Heap.cpp:
* Source/JavaScriptCore/heap/IsoMemoryAllocatorBase.cpp: Removed.
* Source/JavaScriptCore/heap/IsoMemoryAllocatorBase.h: Removed.
* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.cpp:
(JSC::StructureAlignedMemoryAllocator::tryAllocateAlignedMemory):
(JSC::StructureAlignedMemoryAllocator::freeAlignedMemory):
(JSC::StructureMemoryManager::StructureMemoryManager):
(JSC::IsoMemoryAllocatorBase::freeAlignedMemory):
(JSC::StructureAlignedMemoryAllocator::initializeStructureAddressSpace):
(JSC::StructureAlignedMemoryAllocator::StructureAlignedMemoryAllocator): Deleted.
(JSC::StructureAlignedMemoryAllocator::~StructureAlignedMemoryAllocator): Deleted.
(JSC::StructureAlignedMemoryAllocator::tryMallocBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::freeBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::commitBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::decommitBlock): Deleted.
* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.h:

Canonical link: <a href="https://commits.webkit.org/293025@main">https://commits.webkit.org/293025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5854798ed4e5c56afba9d375aa70394a60c355ec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97667 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17292 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7509 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102773 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48196 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17586 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74388 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31583 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13322 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88300 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54737 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13091 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6202 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47638 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/90343 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83110 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104775 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/96290 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24747 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18046 "Found 2 new test failures: ipc/send-gpu-GetShareableBitmap-RemoteRenderingBackend.html pdf/annotations/radio-buttons-select-second.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83441 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25119 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84446 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82872 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27435 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5119 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18344 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15798 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24708 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29877 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/119915 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24530 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33660 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27844 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26104 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hyjorc1 reviewed (approved):
r=me if EWS is happy.

Constellation committed:
[JSC] Use libpas for Structure Heap
https://bugs.webkit.org/show_bug.cgi?id=290685
rdar://148155464

Reviewed by Yijia Huang.

This patch makes StructureAlignedMemoryAllocator much simpler.

1. We remove IsoAlignedMemoryAllocator parts and just let StructureMemoryManager
   allocate and free memory.
2. In libpas-enabled environment, libpas governs Structure heap, and we
   just use libpas functions to allocate and free memory.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/heap/Heap.cpp:
* Source/JavaScriptCore/heap/IsoMemoryAllocatorBase.cpp: Removed.
* Source/JavaScriptCore/heap/IsoMemoryAllocatorBase.h: Removed.
* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.cpp:
(JSC::StructureAlignedMemoryAllocator::tryAllocateAlignedMemory):
(JSC::StructureAlignedMemoryAllocator::freeAlignedMemory):
(JSC::StructureMemoryManager::StructureMemoryManager):
(JSC::IsoMemoryAllocatorBase::freeAlignedMemory):
(JSC::StructureAlignedMemoryAllocator::initializeStructureAddressSpace):
(JSC::StructureAlignedMemoryAllocator::StructureAlignedMemoryAllocator): Deleted.
(JSC::StructureAlignedMemoryAllocator::~StructureAlignedMemoryAllocator): Deleted.
(JSC::StructureAlignedMemoryAllocator::tryMallocBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::freeBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::commitBlock): Deleted.
(JSC::StructureAlignedMemoryAllocator::decommitBlock): Deleted.
* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.h:

Canonical link: https://commits.webkit.org/293025@main
